src/cursor/cur_hs.c

```diff
@@ -670,14 +670,18 @@ __curhs_search_near(WT_CURSOR *cursor, int *exactp)
             WT_ERR(ret);
             /*
              * We find an update when walking forwards. If initially we land on the same key as the
-             * specified key, exact will be 0 and we should return that. If it is not visible, we
+             * specified key, we need to compare the keys to see where we are now. Otherwise, we
              * must have found a key that is larger than the specified key.
              */
-            *exactp = exact;
+            if (exact == 0) {
```

Comment:
I made a mistake before to think that it is OK to remove this key comparison. Actually, it is still needed as we may end up on a larger key if the initial key is not visible.

I can make further code changes to remove this key comparison but I need to pass a flag to the  `__curhs_prev_visible` and `__curhs_next_visible` functions to know whether the cursor is moved in these functions. I think we should just compare the keys here instead of introducing this extra complication.

Reply:
Can you create a ticket to review this in future? It will be great if we can eliminate this comparison.

Reply:
Created WT-7347